File: 이승민/[20.03.24]2331.py

if __name__ == "__main__":
    a, p = map(int, input().split())
    seq = list()
    seq.append(a)

    count = 0
    while True:
        a = str(seq[-1])
        sum_value = 0
        l = len(str(a))
        for i in range(l): sum_value += int(a[i]) ** p
        if sum_value in seq:
            print(seq.index(sum_value))
            break

        seq.append(sum_value)

File: 이승민/[20.03.24]2331.cpp
#include <iostream>
#include <vector>
#include <cmath>
#include <algorithm>

using namespace std;

int main(){
    int a = 0, p = 0;
    cin >> a >> p;
    vector<int> seq;
    seq.push_back(a);

    int count = 0;
    while(true){
        int sum_value = 0;
        string last = to_string(seq.back());
        for (int i = 0; i < last.size(); i++)
        {
            int num = stoi(last.substr(i, 1));
            sum_value += pow(num, p);
        }
        
        for(int i = 0; i < seq.size(); i++){
            if( seq.at(i) == sum_value){
                cout << i << "\n";
                return 0;
            }
        }

        seq.push_back(sum_value);
    }
    
    return 0;
}
